File: server/core/src/test/scala/org/infinispan/server/core/MarshallingTest.scala
package org.infinispan.server.core

import org.testng.annotations.Test
import org.testng.AssertJUnit._
import java.util

/**
 * Marshalling test for server core classes.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
@Test(groups = Array("functional"), testName = "server.core.MarshallingTest")
class MarshallingTest extends AbstractMarshallingTest {

   def testMarshallingBigByteArrayValue {
      val cacheValue = getBigByteArray
      val bytes = marshaller.objectToByteBuffer(cacheValue)
      val readValue = marshaller.objectFromByteBuffer(bytes).asInstanceOf[Array[Byte]]
      assertTrue(util.Arrays.equals(readValue, cacheValue))
   }

}

File: server/core/src/test/java/org/infinispan/server/core/MarshallingTest.java
package org.infinispan.server.core;

import org.testng.AssertJUnit;
import org.testng.annotations.Test;

import java.io.IOException;
import java.util.Arrays;


/**
 * Marshalling test for server core classes.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
@Test(groups = "functional", testName = "server.core.MarshallingTest")
public class MarshallingTest extends AbstractMarshallingTest {
   public void testMarshallingBigByteArrayValue() throws IOException, InterruptedException, ClassNotFoundException {
      byte[] cacheValue = getBigByteArray();
      byte[] bytes = marshaller.objectToByteBuffer(cacheValue);
      byte[] readValue = (byte[]) marshaller.objectFromByteBuffer(bytes);
      AssertJUnit.assertTrue(Arrays.equals(readValue, cacheValue));
   }
}